akka-stream/src/main/scala/akka/stream/impl/PhasedFusingActorMaterializer.scala

```diff
@@ -723,4 +730,73 @@ final class ProcessorModulePhase(materializer: PhasedFusingActorMaterializer) ex
   override def takePublisher(slot: Int, publisher: Publisher[Any]): Unit = publisher.subscribe(processor)
 
   override def onIslandReady(): Unit = ()
+}
+
+object TlsModuleIslandTag extends IslandTag
+
+final class TlsModulePhase(materializer: PhasedFusingActorMaterializer) extends PhaseIsland[TlsModulePhase.MaterializedInfo] {
+  import TlsModulePhase._
+
+  def name: String = "TlsModulePhase"
+
+  def materializeAtomic(mod: AtomicModule[Shape, Any], attributes: Attributes): (MaterializedInfo, Any) = {
+    val es = materializer.effectiveSettings(attributes)
+    println(mod)
+    val tls = mod.asInstanceOf[TlsModule]
```

Comment:
Here's the layout:

```
--- Materializing layout:
 | push attr Attributes(List(Name(headSink), InputBuffer(1,1)))
 |  |  | materialize GraphStage(HeadOptionStage) [28ac7cef] []
 |  | transform mat
 | pop attr
 | push attr Attributes(List())
 |  | materialize GraphStage(DropWhile) [52dab3fe] [-1]
 |  | pop attr
 |  | pop mat
 |  | push attr Attributes(List())
 |  | materialize GraphStage(akka.stream.io.TlsSpec$Timeout@7c40e710) [65510559] [-1]
 |  | pop attr
 |  | pop mat
 |  | push attr Attributes(List())
 |  | materialize GraphStage(Scan) [3740b773] [-1]
 |  | pop attr
 |  | pop mat
 |  | push attr Attributes(List())
 |  | materialize GraphStage(Collect) [3002a726] [-1]
 |  | pop attr
 |  | pop mat
 |  | push attr Attributes(List())
 |  | materialize GraphStage(Map(class akka.stream.io.TlsSpec$$anonfun$1$$anonfun$7)) [5d1407b2] [-1]
 |  | pop mat
 |  |  | push NotUsed
 |  | pop attr
 |  | pop mat
 |  | push attr Attributes(List())
 |  | materialize GraphStage(akka.stream.io.TlsSpec$$anonfun$1$$anonfun$apply$mcV$sp$6$$anonfun$apply$6$$anonfun$apply$7$$anonfun$apply$8$$anon$1@14ded3e9) [26970367] [-1]
 |  | pop attr
 |  | pop mat
 |  | push attr Attributes(List())
 |  | enter island akka.stream.impl.TlsModuleIslandTag$@6c3b70ec
 |  |  |  | materialize TlsModule(IgnoreComplete) [31406dc3] [3, -1]
 |  |  |  | enter island akka.stream.impl.TlsModuleIslandTag$@6c3b70ec
 |  |  |  |  | materialize TlsModule(IgnoreComplete) [4c14d5ce] [-1, 3]
 |  |  |  | pop mat
 |  |  |  | push attr Attributes(List())
 |  |  |  |  | materialize GraphStage(Map(class akka.stream.io.TlsSpec$$anonfun$1$PayloadScenario$1$$anonfun$flow$1)) [52066b10] [-2]
 |  |  |  |  | pop mat
 |  |  |  |  |  | push NotUsed
 |  |  |  |  | pop attr
 |  |  |  |  | pop mat
 |  |  |  |  | push attr Attributes(List())
 |  |  |  |  | materialize GraphStage(Map(class akka.stream.io.TlsSpec$$anonfun$1$$anonfun$7)) [5d1407b2] [-1]
 |  |  |  |  | pop mat
 |  |  |  | push NotUsed
 |  |  | pop mat
 |  | pop attr
 |  | pop mat
 |  | push attr Attributes(List(Name(iterableSource)))
 |  |  | push attr Attributes(List())
 |  |  | materialize GraphStage(StatefulMapConcat) [3025ab22] [-6]
 |  |  | pop attr
 |  |  | pop mat
 |  | materialize GraphStage(SingleSource(List(SendBytes(ByteString(104, 101, 108, 108, 111)), SendBytes(ByteString()), SendBytes(ByteString(32, 119, 111, 114, 108, 100))))) [0239e690] [-1]
 | pop attr
pop mat
```

I'll try to simplify the flow tomorrow to make it simpler to debug.